src/ngraph/runtime/gpu/gpu_util.hpp

```diff
@@ -52,6 +52,16 @@
         }                                                                                          \
     } while (0)
 
+#define CUDNN_SAFE_CALL(func)                                                                      \
```

Comment:
Why is this a macro and not a function?

Reply:
My understanding is Macro is easier to take any function as an input. It will be much harder to do with a function. I agree, at some point we need to have our own error handle.

Reply:
Also, a lot of frameworks implement these as defines (e.g. https://github.com/NervanaSystems/ngraph-mxnet/blob/3857caef796fc9d4c3df4ea5e9f83f8af3f089ae/src/common/cuda_utils.h#L175) so that they can be wrapped up with ifndef NDEBUG so that the asserted code is only checked for debug builds. Something we should add.